aspnetcore/client-side/import-export-interop.md

```diff
@@ -0,0 +1,394 @@
+
+To export a .NET method so it can be called from JS, use the `JSExportAttribute`.
+
+In the following example, the `Greeting` method returns a string that includes the result of calling the `GetHRef` method. As shown earlier, the `GetHref` C# method calls into JS for the `window.location.href` function from the `main.js` module. `window.location.href` returns the current page address (URL). In an app generated from either the `wasmbrowser` or `wasmconsole` templates, `Greeting` is in the `Program.cs` file with the partial class namespace `MyClass`:
+
+```csharp
+[JSExport]
+internal static string Greeting()
+{
+    var text = $"Hello, World! Greetings from {GetHRef()}";
+    Console.WriteLine(text);
+    return text;
+}
+```
+
+The `dotnet.js` file is used to create and start the .NET WebAssembly runtime. `dotnet.js` is generated as part of the build output of the app and found in the `AppBundle` folder:
+
+> :::no-loc text="bin/{BUILD CONFIGURATION}/{TARGET FRAMEWORK}/browser-wasm/AppBundle":::
+
+The `{BUILD CONFIGURATION}` placeholder is the build configuration, and the `{TARGET FRAMEWORK}` placeholder is the target framework.
+
+To integrate with an existing app, copy the contents of the `AppBundle` folder so that it can be served along with the rest of the app. For production deployments, publish the app with the `dotnet publish -c Release` command in a command shell and deploy the `AppBundle` folder with the app.
+
+In the following example, the app contains an HTML file (`.htm`/`.html`) that loads a JS file (`.js`), which imports `dotnet.js` and starts the .NET runtime:
+
+```javascript
+import { dotnet } from './dotnet.js'
+
+await dotnet.run();
+```
+
+## Experimental workload and project templates
+
+To demonstrate the JS interop functionality and obtain JS interop project templates, install the `wasm-experimental` workload:
+
+```dotnetcli
+dotnet workload install wasm-experimental
+```
+
+The `wasm-experimental` workload contains two project templates: `wasmbrowser` and `wasmconsole`. These templates are experimental at this time, which means the developer workflow for the templates hasn't been fully designed. For example, these templates don't run in Visual Studio at this time. The .NET and JS APIs used in the templates are supported in .NET 7 and provide a foundation for using .NET on :::no-loc text="WASM"::: from JS.
+
+The templates can be examined in reference source:
+
+[`templates` assets in the `dotnet/runtime` GitHub repository](https://github.com/dotnet/runtime/tree/main/src/mono/wasm/templates/templates).
+
+> [!NOTE]
+> The preceding link to the .NET reference source loads the repository's default branch (`main`), which represents the current development for the next release of .NET. To select a tag for a specific release, use the **Switch branches or tags** dropdown list.
+
```

Comment:
Instead of trying to read the template source on GitHub the user can just use the templates to create a project.

```suggestion
```